ruff_lsp/server.py

```diff
@@ -679,99 +968,166 @@ async def resolve_code_action(params: CodeAction) -> CodeAction:
         f"{CodeActionKind.SourceOrganizeImports.value}.ruff",
     ):
         # Generate the "Ruff: Organize Imports" edit
-        results = await _fix_document_impl(document, only="I001")
-        params.edit = _create_workspace_edits(document, results)
+        params.edit = await _fix_document_impl(document, only="I001")
 
     elif settings["fixAll"] and params.kind in (
         CodeActionKind.SourceFixAll,
         f"{CodeActionKind.SourceFixAll.value}.ruff",
     ):
         # Generate the "Ruff: Fix All" edit.
-        results = await _fix_document_impl(document)
-        params.edit = _create_workspace_edits(document, results)
+        params.edit = await _fix_document_impl(document)
 
     return params
 
 
 @LSP_SERVER.command("ruff.applyAutofix")
-async def apply_autofix(arguments: tuple[TextDocument]):
+async def apply_autofix(ls: server.LanguageServer, arguments: tuple[TextDocument]):
```

Comment:
How do we now have the `ls` here?

Reply:
Ah, it's not strictly required. We can either use the `ls` instance or the global `LSP_SERVER`, they both are the same.